src/store/reader.rs

```diff
@@ -74,24 +130,30 @@ impl StoreReader {
         self.data.slice(checkpoint.byte_range.clone()).read_bytes()
     }
 
+    /// Loads and decompresses a block.
+    ///
+    /// Advanced API. In most cases use [get](Self::get).
     fn read_block(&self, checkpoint: &Checkpoint) -> io::Result<Block> {
-        if let Some(block) = self.cache.lock().unwrap().get(&checkpoint.byte_range.start) {
-            self.cache_hits.fetch_add(1, Ordering::SeqCst);
-            return Ok(block.clone());
+        if let Some(block) = self.cache.get_from_cache(checkpoint.byte_range.start) {
+            return Ok(block);
         }
 
-        self.cache_misses.fetch_add(1, Ordering::SeqCst);
-
         let compressed_block = self.compressed_block(checkpoint)?;
+
+        self.decompress_block(compressed_block, checkpoint.byte_range.start)
```

Comment:
Can we put the put_cache logic in this function (as opposed to putting it into decompress_block)?

The idea is that the caching all happen in the same function.